appformer-client-api/src/main/java/org/appformer/client/stateControl/registry/CommandRegistry.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.appformer.client.stateControl.registry;
+
+import java.util.List;
+
+/**
+ * Represents a registry of commands to be used on the StateControl engine.
+ *
+ * @param <C> anything that can be considered a Command
+ */
+public interface CommandRegistry<C> {
```

Comment:
in fact this interface describes a generic `Registry`... I mean, even being called `CommandRegistry`, the `C` parameter is a generic type, which does not extends from any other types as well.
So just as a suggestion, as we're moving this to the fundation codebase, we can renane it just as `Registry<O>`, it can contain commands, or whatever, also we could remove the term "command" from the methods below.
Anyway just as a proposal, but I see no reason why it describes a registry of commands, nothing actually is realted to the Command API...

Reply:
I think I get your point. Replying as was one of assumed **Command** substring properly used in names of classes.

I think you are correct we can remove **Command** form name of this interface and keep it just in real implementations which are:
```java
XYZCommandRegistry implements Registry<Command>
```

Reply:
Ok